src/wallet/wallet.cpp

```diff
@@ -3888,11 +3859,42 @@ bool CWallet::SelectStakeCoins(std::list<std::unique_ptr<ZerocoinStake> >& listI
         if (meta.nVersion < CZerocoinMint::STAKABLE_VERSION)
             continue;
         if (meta.nHeight < chainActive.Height() - nRequiredDepth) {
-            std::unique_ptr<ZerocoinStake> input(new ZerocoinStake(meta.denom, meta.hashStake));
+            std::unique_ptr<StakeInput> input(new ZerocoinStake(meta.denom, meta.hashStake));
             listInputs.emplace_back(std::move(input));
         }
     }
 
+    /*
+     * void AnonWallet::AvailableAnonCoins(std::vector<COutputR> &vCoins, bool fOnlySafe, const CCoinControl *coinControl,
+        const CAmount& nMinimumAmount, const CAmount& nMaximumAmount, const CAmount& nMinimumSumAmount,
+        const uint64_t& nMaximumCount, const int& nMinDepth, const int& nMaxDepth, bool fIncludeImmature)
+     */
+
+    //todo time based activation here
+    std::vector<COutputR> vRctCoins;
+    pAnonWalletMain->AvailableAnonCoins(vRctCoins);
+    for (const COutputR& rctOut : vRctCoins) {
+        const CTransactionRecord& txrecord = rctOut.rtx->second;
+
+        //Check transaction age
+        if (GetAdjustedTime() - txrecord.nBlockTime > nStakeMinAge)
+            continue;
+
+        //Check output
+        const COutputRecord* pout = txrecord.GetOutput(rctOut.i);
+        if (pout->GetAmount() < Params().MinimumStakeQuantity())
+            continue;
+
+        CTransactionRef ptx;
+        uint256 hashblock;
+        if (!GetTransaction(rctOut.txhash, ptx, Params().GetConsensus(), hashblock, true))
+            continue;
+
+        //Create stake input and add to list of stakable inputs
+        std::unique_ptr<StakeInput> input(new RingCtStakeCandidate(this, ptx, rctOut.GetOutpoint(), pout));
+        listInputs.emplace_back(std::move(input));
+    }
+
     LogPrint(BCLog::BLOCKCREATION, "%s: FOUND %d STAKABLE ZEROCOINS\n", __func__, listInputs.size());
```

Comment:
Change `ZEROCOINS` to `COINS`

Reply:
> Change `ZEROCOINS` to `COINS`

should be Change `ZEROCOINS` to `TXOs`, or something non denominational since it's not the total value, but rather total number of inputs